fighthealthinsurance/ml/ml_router.py

```
@@ -131,31 +129,34 @@ def full_find_citation_backends(self, use_external=False) -> list[RemoteModelLik
         # Only use Perplexity models for citations
         if "sonar-reasoning" in self.models_by_name:
-            citation_models.extend(self.models_by_name["sonar-reasoning"])
+            return self.cheapest("sonar-reasoning")
         if "sonar" in self.models_by_name:
-            citation_models.extend(self.models_by_name["sonar"])
+            return self.cheapest("sonar")
 
-        return citation_models
+        return []
+
+    def cheapest(self, name: str) -> list[RemoteModelLike]:
```

Ruff (0.8.2)

`RemoteModelLike` may be undefined, or defined from star imports

(F405)